### PR TITLE
support httpx streaming

### PR DESCRIPTION
**Pull request by stevenbunting**
image
This is kinda the basis for why supporting capturing streamed responses is difficult with httpx

If the response is not being streamed, it always gets read, so there's an obvious spot for the patch. For streamed responses the same isn't true. It seems like some of the benefit of streaming is deciding how much to read into memory at any given time, so the raw response object is returned. An end user could call `.read()` on that object (and our current implementation would cache it) but there are other entry points to reading the data within, one of which is to iter over it.

While we could call `.read()` ourselves on that response object and cache it, it's not clear to me we want to do that, as it seems to break one of the fundamental "purposes" of streaming a response. What seems more reasonable is hooking into the iterator and "following along" if a user chooses to iterate the response instead.

TL;DR
- There is a known weakness where if the user stops iterating before parsing the whole response, we won't capture it. The hanging request will stay on the request cache until a force flush occurs. This is to avoid reading all of a streamed response into memory before the surrounding code decides to

### CODE REVIEW

**Review by aklarfeld — APPROVED** [review]
Awesome - thanks for digging into this, Steve.
